scripts/copper/lithium/prepare.py

```diff
@@ -0,0 +1,198 @@
+from pseudopotential_calculator.calculations.adsorbate import (
+    get_adsorbate_position,
+    get_scaled_basis_vectors,
+    get_slab_with_adsorbate,
+    get_top_layer_z_vector,
+    get_top_position,
+    prepare_adsorbate,
+    prepare_slab_with_adsorbate,
+)
+from pseudopotential_calculator.calculations.slab import (
+    get_surface,
+)
+from pseudopotential_calculator.castep import (
+    Castep,
+    CastepConfig,
+    CastepParams,
+    get_calculator,
+    load_calculator_atoms,
+)
+from pseudopotential_calculator.hpc import (
+    prepare_calculator_with_submit_script,
+    prepare_submit_all_script,
+)
+from pseudopotential_calculator.scripting import maybe_copy_files_to_hpc
+from pseudopotential_calculator.util import (
+    prepare_clean_directory,
+)
+
+if TYPE_CHECKING:
+    from ase import Atoms
+
+ENERGY_MAP_PATH = Path("data/copper/lithium/energy_map")
+VACUUM_LAYER_PATH = Path("data/copper/lithium/vacuum_layer")
+SLAB_WIDTH_PATH = Path("data/copper/lithium/width")
+
+
+def _prepare_vacuum_layer_convergence(atoms: Atoms, data_path: Path) -> None:
+    calculators = list[Castep]()
+    prepare_clean_directory(data_path)
+
+    adsorbate_height = 1.0
+
+    for n_vacuum_layer in range(1, 9):
+        slab = get_surface(
+            atoms,
+            (1, 1, 1),
+            n_fixed_layer=4,
+            n_free_layer=3,
+            n_vacuum_layer=n_vacuum_layer,
+            adsorbate_height=adsorbate_height,
+        )
+
+        slab_position = get_top_position(slab)
+        adsorbate_position = get_adsorbate_position(
+            slab_position,
+            adsorbate_height,
+        )
+        li_atom = prepare_adsorbate("Li", adsorbate_position)
+
+        slab_with_adsorbate = get_slab_with_adsorbate(li_atom, slab)
+
+        config = CastepConfig(
+            data_path / f"slab_{n_vacuum_layer}_vacuum_layer",
+            "slab",
+        )
+        params = CastepParams(n_k_points=(12, 12, 1), xc_functional="WC")
+        calculator = get_calculator(slab_with_adsorbate, params, config)
+        prepare_calculator_with_submit_script(calculator)
+
+        calculators.append(calculator)
+    prepare_submit_all_script(calculators, data_path)
+    maybe_copy_files_to_hpc(data_path, PosixPath(data_path.as_posix()))
+
+
+def _prepare_slab_width_convergence(
+    atoms: Atoms,
+    data_path: Path,
+) -> None:
+    calculators = list[Castep]()
+    prepare_clean_directory(data_path)
+    adsorbate_height = 1.0
+
+    slab = get_surface(
+        atoms,
+        (1, 1, 1),
+        n_fixed_layer=4,
+        n_free_layer=3,
+        n_vacuum_layer=5,
+        adsorbate_height=adsorbate_height,
+    )
+
+    slab_position = get_top_position(slab)
+    adsorbate_position = get_adsorbate_position(
+        slab_position,
+        adsorbate_height,
+    )
+    li_atom = prepare_adsorbate("Li", adsorbate_position)
+
+    for width in range(1, 9):
+        slab_with_adsorbate = prepare_slab_with_adsorbate(li_atom, slab, width)
```

Comment:
this should deal with the whole relative coordinate behaviour you are doing in this function